kahuna/public/js/services/panel.js

```diff
@@ -0,0 +1,139 @@
+import angular from 'angular';
+
+export const panelService = angular.module('kahuna.services.panel', []);
+
+panelService.factory('panelService', [ '$rootScope', '$timeout', function ($rootScope, $timeout) {
+
+    class UIPanel {
+        constructor(options) {
+            this.name = options.name;
+            this.initVisible = options.initVisible || false;
+            this.initAvailable = options.initAvailable || false;
+            this.initLocked = options.initLocked || false;
```

Comment:
why do we need to assign `this.initX` rather than `this.visible = options.initVisible || false`?